lib/Sema/TypeCheckAttr.cpp

```diff
@@ -51,7 +51,12 @@ namespace {
     assert(!D->hasClangNode() && "Clang importer propagated a bogus attribute");
     if (!D->hasClangNode()) {
       SourceLoc loc = attr->getLocation();
-      assert(loc.isValid() && "Diagnosing attribute with invalid location");
+      if (!loc.isValid()) {
+        llvm::errs() << "Attribute '";
```

Comment:
I think we need to surround these prints with `#ifndef NDEBUG`, etc, so they don't show up if asserts are off. Otherwise, 👍🏼

Reply:
Thanks, yes that's nice to not impact release builds - thanks